### DISCUSSION TITLE
Locations divorced from online meetings make them harder to find

### DISCUSSION THREAD

**Discussion by alex-m-dc** [code4recovery/12-step-meeting-list#437] [General]
Hello,
By totally removing the location from an online meeting (Except for a regional approximate) makes it harder to find meetings.

Say someone knows a meeting as "the grace church 6pm" meeting, one of many held there, but they have different names and some are in-person and some are online. Yes, you can still search for the location and see all held there but when viewing a standard listing of upcoming meetings they won't see that location name and it may confuse them.

And... clubs... this is really bad for them when someone is used to looking up the "Dupont Circle Club 6pm" even though it's called "Beginner Basics" if the club name isn't there... they can't find it.

Please bring back location names for online meetings in the list - you could still prevent the street address from showing up... 

And not having them greyed out would be good - I've had people complain it makes the location look closed even though there are in-person meetings being held there, just that one is online only.

THANKS!!!

--Alex M
267-231-2174

**Comment by Grateful-Viv** [comment]
I agree with you Alex, but for different reasons. 

What they decided to do is keep the location name if it's in person (e.g. Dupont Circle Club), but if it's an online meeting to use group name if there is one, and if there isn't then it's empty. You can see the conversation over in this [thread](https://github.com/code4recovery/tsml-ui/issues/119) which started before this WordPress plugin was updated to v3.11.0.

**Comment by tech2serve** [comment]
> Hello,
> By totally removing the location from an online meeting (Except for a regional approximate) makes it harder to find meetings.
> 

How so? If a meeting is online only, then nobody should be trying to drive to it. Instead, they click on the online link and attend that way.

> Say someone knows a meeting as "the grace church 6pm" meeting, one of many held there, but they have different names and some are in-person and some are online. Yes, you can still search for the location and see all held there but when viewing a standard listing of upcoming meetings they won't see that location name and it may confuse them.
> 

Any in-person meeting, be it only physical or be it a hybrid, should show the address as that is necessary to allow a user to drive to the meeting.

> And... clubs... this is really bad for them when someone is used to looking up the "Dupont Circle Club 6pm" even though it's called "Beginner Basics" if the club name isn't there... they can't find it.
> 

Same here. If it is not possible for the person to attend the meeting in person, we don’t want to show the address in order to avoid confusing them.

Let’s take this screen cap as an example from our demo site:

![119AF6D1-5FE8-4097-8CC9-3B246BA3EF45](https://user-images.githubusercontent.com/31904316/125180586-7fd39b00-e1b0-11eb-80fa-2281befe111d.jpeg)

Note that for the Alano Club West the meeting can be attended in person or online (i.e., it is a hybrid meeting). Therefore, the meeting list shows both the name of the club/group and the address. 

Now look at Sunrisers. This is a good example of where there are two different meetings being held at the same time; one meeting is in person and the other is online. In this case, it isn’t a hybrid meeting (i.e., neither the chairperson nor the attendees interact between meetings).

For the meeting in person, the location and the address are shown so members can drive to and attend the meeting.

For the online version of the meeting, the group name is still provided along with the flag making it clear the meeting is online.

One of our goals is to get meeting lists back to being more focused on information necessary to how a member can attend any given meeting. 

Hopefully this helps clear up our logic and approach.

Yis, Tim

**Reply by Grateful-Viv** [reply]
In your example above for Sunrisers, putting the group name for their 630 online meeting in place of location, but not putting that for the 630 in-person meeting is confusing. Is it the same group? How does putting the group name help the person looking for a meeting? I believe in the case of the online meeting, just putting “online” only is clearer

Version 3.11 is still displaying in the detail page all meetings for the location and that is the case whether meetings are inactive, online, in-person or even part of the same group.

**Comment by alex-m-dc** [comment]
Not group name, location name in the location column. Leave the address
line off, keep the map off, just need that location listed or I can't
upgrade.

**Reply by tech2serve** [reply]
What is your site? And what meetings are problematic?

If the meeting location is closed, why should the site show the location?

**Comment by alex-m-dc** [comment]
As I said, many times people know meetings by their location not their
name. "The x club 6pm"

Aa-dc.org

**Reply by tech2serve** [reply]
With v3.11, if a meeting is in person or hybrid, the location should show on the meeting list. 

It is only for natively online meetings that the group name will be shown because having a location doesn't make sense and could be confusing to new comers. 

And if there is no Group information, the meeting list simply shows "Online" under the location/group column.

We believe this approach makes more sense for the majority of users, and especially newcomers. The new UX strives to make a clearer distinction between online and in-person meetings, and it makes it clearer to the user when they cannot attend at a location. We believe including a location name for a natively online meeting muddies this distinction. Most people use the meeting name for referencing meetings, and so we continue to show this. Secondarily, people use either the location name (fi attending at that location) or a group name if online.

The beauty of open source software is that you can fork and maintain your own version that you can use and deploy on your site.

**Reply by joshreisner** [reply]
If this column is empty @alex-m-dc then one workaround is to populate the group field with, for example, "Dupont Circle Club" for all meetings at the Dupont Circle Club. There is a checkbox that appears to facilitate marking all meetings at a location with a single update. It's an even quicker update with a CSV.

**Reply by Grateful-Viv** [reply]
The problem with marking all group fields with the location name for online meetings because you want the meeting grid list to populate with the data you want to display is that it takes away the group name entry if it does indeed have one. It defeats the purpose of putting in the group name (and the district number) and then any group notes (e.g. “group business meetings are the last Saturday of the month”). 

We have some same groups doing online meetings and in-person meetings. At the same time. Not hybrid. Separate meetings. Yet looking at the list, it appears only half the meetings from that group belong to the group.

**Reply by joshreisner** [reply]
@Grateful-Viv I would imagine that you would only do this for group entries where there wasn't already a group name in place.

**Reply by Grateful-Viv** [reply]
Many of our meetings belong to a group. See screenshot below for what is confusing people since I updated the plug-in. I did make CSS modifications to it because I didn’t like the empty cells. But take a look at Bethany Early Birds group. They meet at 8 AM and have an online meeting and also have an in person meeting. 

![01AEBB0D-D633-4D44-9C19-3830D23EE496](https://user-images.githubusercontent.com/62247421/125342283-b29da080-e322-11eb-868d-6cc65b484cd7.jpeg)

In an ideal world, if I could get my way that is 😁, I wouldn’t show Group at all. And for the online meeting it would show the location that would either be crossed off or just put the word online there. 

I get the logic for online meetings that never met in person but even some of those meetings actually became a registered group. And for those I would love it if it just said online instead of having the group name. The name of the group is not helping anybody locate the meeting whether it’s virtual or by a street address. 

What Alex is talking about is a little different because people in DC are used to referring to the triangle club at 6 PM meeting or the Westside club at 2 PM. So I didn’t really mean to hijack his thread but it is a similar issue in that putting group name in the location area isn’t helping people locate meetings that they want to go to.

**Comment by alex-m-dc** [comment]
An impediment to using the group solution (i.e. put "grace church" as the
group name so it appears in the list) is that we are then associating all
meetings at that location with the same "group" and they are separate,
individual groups.


Also, we will NOT add anything a groups record which they don't explicitly
tell us to add. So even if we did want to try this solution, we would have
to contact every group/meeting.

How about the "online meeting" indicator goes under the name of the
location in that column? If there is a name and an approximate location or
street address and it's online only, then the street address is replaced by
"online meeting"...

**Reply by joshreisner** [reply]
My point was simply that you can to use these group names to fill the column however you want. I am very much not recommending putting church names in the group field (this was the whole point of making this change, to get location names out of that column) but because you said you wanted location information there I suggested a way you could achieve that outcome.

I think the bigger issue here is that WAIA is using this field in a way we hadn't expected. For example [this online Dupont Circle Club meeting](https://aa-dc.org/meetings/dupont-circle-club-6) has a group name of DCC 9:30 AM. I'm guessing that this has to do with the unique way they organize their service structure. Previously I would have expected that most locations like this that have 85 meetings / week would have a single group. 

Likewise [this online Westside Club meeting](https://aa-dc.org/meetings/attitude-and-action-2) has a group name that's the same as the meeting name, while other online meetings at the Westside Club don't have group names.

I would think it would be more valuable to meeting-finding not to try to reflect these organizational distinctions, but to mark these all as Westside Club and Dupont Circle Club, but I hear you that it's not possible in this case.

I'm sure the TSML maintainers will discuss this and decide on a path forward. My investment in it comes from wanting to support TSML UI and the Meeting Guide specification. 

TSML UI doesn't use the word Online in the listings the same way TSML does (eg https://aasanjose.org/aa-meetings) but perhaps a partial solution is if a meeting a) does not have a group, and b) is associated with a specific physical location, and c) is marked Location Temporarily Closed, then we show the location name in that column.

**Reply by Grateful-Viv** [reply]
In this plug-in version, group names are displayed only if the admin meeting entered a group name and meeting is either online or inactive. It is dimmed out too. If same group also holds a hybrid or in-person meeting, the location info is displayed and not the group. 

I’m not in favor of displaying group names in this manner. It shouldn’t be an either or.  But if you must, it should be across all attendance options. The same logic, I believe should also apply to TSML-ui.

**Reply by alex-m-dc** [reply]
@joshreisner this is a great example of how things are organized differently - there are several strong groups at my clubhouse, each with their own intergroup and GSO representative  - some are just one meeting, some are 7 days a week and now 2X with an online and an in-person version. 

There are churches and other locations which have multiple of meetings a week, each is considered a separate group.

I'm not sure with all the back and forth what the decision here is - or if its' still pending or if I've been able to adequately explain why this is needed in my area. 

Thank you

**Comment by alex-m-dc** [comment]
@tech2serve The issue has been closed? This is a big deal for my area and for the issue to be summarily closed without notification is rough...

@joshreisner @Grateful-Viv  What are my options here? We can't update when the location name, not address!! isn't displayed.

IMHO it should be the site's choice if the location shouldn't display - they can remove it and a blank location won't show up.

Are the admins really that worried about people showing up at a location for an online meeting they want to take that choice away from the people providing the service?

**Reply by joshreisner** [reply]
@alex-m-dc , @tech2serve didn't mean anything by it, that's just what happens when an issue is moved to a discussion. He brought it to our Slack group to discuss internally yesterday, he's hoping to bring in other opinions about it.

I would not expect an immediate answer though. We are unpaid volunteers, and this is a tough gig sometimes.

**Reply by alex-m-dc** [reply]
Thank you! I did feel a bit dismissed - but realize I should read less into brevity, and more into the fact that god's got this and the spirit of AA will persist!

Let me know how I can help with the discussion - we're doing fine with our custom "in person" type at the moment - so the only urgency is to use the great features ya'll built into 3.11.

**Reply by tech2serve** [reply]
I promise you're not being dismissed. :-)

It is simply that we recognize different webmasters want to do things their way (and perhaps the way they've always done it), and we are striving to hit a solution that is best for a majority of users, and that we believe best serves the newcomers. We don't have the depth of bench to write customizations for every site.

We are approaching a 60% adoption of 3.11, and frankly the negative comments have been minimal...well within the bounds of what I would expect for rolling out a pretty major change. This tells me that most of the users are happy with what we've done.

This doesn't mean the topic or internal discussion has ended. We developers will continue to monitor and evaluate the UX.

Yis, Tim

**Comment by Alex-M-WAIA-DC** [comment]
Hello,
Adding to this thread after a tech committee meeting where it was discussed how important it is in our area for the location name to be part of an online meeting's listing.

We have several clubs in the area and one of them is a standout for people of color: the Metropolis Club. Newcomers need to be able to find meetings associated with that club (online or not!) as easily as possible - and without the club name in the standard list, that will be difficult.

Currently (3.10) this is what they see for a temp closed meeting with a online version:
![image](https://user-images.githubusercontent.com/87840666/126730605-63f6c15d-4422-42e3-b3ed-723dca92a5de.png)
Note the greyed out location and address, which may seem to indicate it's in-person.... so we NEED to upgrade to prevent confusion.

This is how it looks in the new version:
![image](https://user-images.githubusercontent.com/87840666/126730680-99bf66cc-dc0e-4aea-af95-6b14fd95b4d0.png)
No way to know it's a Metropolis club meeting!

Koffee Klatch is it's own group - so setting the group name to the club name would be incorrect  and infeasible for the hundreds of groups/meetings. 

What I request is:
- Location **name** stays.*
- "Online" below the location name changes to "Online Only"
- The address is removed.

* It could be superseded by group name if it exists or, better yet allow location name to be blank... (Not sure if that is ok with the App...)

Is this a reasonable compromise?

--Alex M
(Posting as the technology chairperson of the Washington Area Intergroup Association.)

**Comment by Grateful-Viv** [comment]
If you could show whatever was put in the location field for an online meeting, you could put something like "Metropolis Club Online" and that should be clear to everyone.

In our case for Southern Delaware Intergroup, displaying the group name in place of the location data, serves no purpose and only adds to confusion since a few groups are hosting online meetings and in-person meetings at the same time (not hybrid) and looking at the list gives the impression that only one meeting is part of the group.

**Reply by Grateful-Viv** [reply]
Is this being worked on? It would still be preferable to display whatever the admin keyed into the location field. If the attendance type is inactive and used to meet at a location, then that should be displayed and crossed out. If it's a natively online meeting (let's call this attendance-native as opposed to attendance-inactive, then admins can put whatever they want in the location field or nothing at all. Whatever is in there will display on the meeting list grid.
I tried to create a "group" called "Online-Only" where the meeting was not associated with a group and was online so it would show up in the Location/Group cell, in order to better display this was an online meeting.

To my dismay have come to find out that it changed the contacts to the first instance of this group name since it thinks all these meetings are part of this group. We keep try to keep a record of the internal contacts for being able to reach back out to them if there are issues with their meetings. But now it changed all contacts.

**Comment by Alex-M-WAIA-DC** [comment]
DC is still on 3.10.0 - they cannot update without the ability to show location names (in some form!) in the list. 

@joshreisner @Grateful-Viv @tech2serve

**Reply by tech2serve** [reply]
This was discussed this morning and we understand the benefit of being able to display more information associating a meeting with a particular entity that has multiple groups (i.e., a club or church).

We're still working out the particulars as it must also be clear the meeting is *online only* without cluttering the UI. If you haven't looked at it, [our demo site for the future UI](http://demo.code4recovery.org/tsml-ui) shows what we're thinking long-term; and the beta version of TSML proper is [available here](https://demo.code4recovery.org/meetings).

We'll strive to reach a final consensus of how to make the sausage in the next couple of days, and then we'll write up a ticket for which I'll share the link.

**Reply by Grateful-Viv** [reply]
I'd like to echo the previous request to display in the location field to whatever the admin entered. That should not clutter up the UI at all. You can still leave the address field as an empty cell. The region should also display whatever the admin entered.

**Reply by Grateful-Viv** [reply]
Any movement on getting a consensus on making these changes?

**Comment by alex-m-dc** [comment]
Thank you - this means a lot for us.

If your having any trouble with the UI/UX the new tech chair Sam is copied
here and I have a few years of design under my belt.
